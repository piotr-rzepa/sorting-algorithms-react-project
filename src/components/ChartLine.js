/*
 * Komponent bezklasowy będacy wykresem liniowym z czasami worst,best,average \
 * poszczególnych algorytmów na ich stronach
 */

/* eslint-disable react/prop-types */
import React from 'react';
import {
	LineChart,
	Line,
	Legend,
	Tooltip,
	XAxis,
	YAxis,
	CartesianGrid
} from 'recharts';

export default function CardChart(props) {
	return (
		<div
			style={{
				width: `${props.width}px`,
				height: `${props.height}px`
			}}
			className="card-chart card-chart--big"
		>
			<LineChart
				width={props.width}
				height={props.height}
				data={props.data}
				margin={{ top: 5, right: 30, left: 20, bottom: 5 }}
			>
				<CartesianGrid strokeDasharray="3 3" />
				<XAxis dataKey="size" />
				<YAxis />
				<Tooltip />
				<Legend />
				<Line type="monotone" dataKey="worst" stroke="#FF7A00" />
				<Line type="monotone" dataKey="average" stroke="#6F52ED" />
				<Line type="monotone" dataKey="best" stroke="#28E451" />
			</LineChart>
		</div>
	);
}
