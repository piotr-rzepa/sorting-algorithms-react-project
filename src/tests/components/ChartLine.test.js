import * as React from 'react';
import { shallow } from 'enzyme';
import * as reactRedux from 'react-redux';
import ChartLine from '../../components/ChartLine';

const data = [
	{
		id: 0,
		name: 'Bubble Sort',
		description:
			'Bubble sort, sometimes referred to as sinking sort, is a simple sorting algorithm',
		method: 'Exchanging',
		sortingTime: { best: 0, worst: 0, average: 0 }
	},
	{
		id: 1,
		name: 'Quick Sort',
		description: 'Quicksort is a divide-and-conquer algorithm.',
		method: 'Partitioning',
		sortingTime: { best: 0, worst: 0, average: 0 }
	}
];

describe('<ChartLine />', () => {
	const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');
	const useContextMock = jest.spyOn(React, 'useContext');
	beforeEach(() => {
		useSelectorMock.mockClear();
		useContextMock.mockClear();
	});

	it('renders correctly', () => {
		const dummyDispatch = jest.fn();
		useContextMock.mockReturnValue(dummyDispatch);
		useSelectorMock.mockReturnValue(data);
		const wrapper = shallow(<ChartLine />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should have default light style', () => {
		const dummyDispatch = jest.fn();
		useContextMock.mockReturnValue(dummyDispatch);
		useSelectorMock.mockReturnValue(data);
		const wrapper = shallow(<ChartLine />);
		expect(useContextMock).toHaveBeenCalled();
		expect(wrapper.find('div.card-chart').get(0).props.style).toHaveProperty(
			'backgroundColor',
			'#606060'
		);
	});

	it('should choose correct theme', () => {
		useContextMock.mockReturnValue(false);
		useSelectorMock.mockReturnValue(data);
		const wrapper = shallow(<ChartLine />);
		expect(wrapper.find('div.card-chart--big').get(0).props.style).toHaveProperty(
			'color',
			'black'
		);
	});
});
