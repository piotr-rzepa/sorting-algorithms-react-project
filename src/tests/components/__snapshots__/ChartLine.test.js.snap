// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartLine /> renders correctly 1`] = `
<div
  className="card-chart card-chart--big card--line chart-visualize"
  style={
    Object {
      "backgroundColor": "#606060",
      "boxShadow": "2px 2px 2px black",
      "color": "white",
    }
  }
>
  <ResponsiveContainer
    debounce={0}
    height="100%"
    width="100%"
  >
    <LineChart
      barCategoryGap="10%"
      barGap={4}
      layout="horizontal"
      margin={
        Object {
          "bottom": 5,
          "left": 20,
          "right": 30,
          "top": 5,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
      syncMethod="index"
    >
      <CartesianGrid
        fill="none"
        horizontal={true}
        horizontalFill={Array []}
        horizontalPoints={Array []}
        stroke="#ccc"
        strokeDasharray="3 3"
        vertical={true}
        verticalFill={Array []}
        verticalPoints={Array []}
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="size"
        domain={
          Array [
            0,
            "auto",
          ]
        }
        height={30}
        hide={false}
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        domain={
          Array [
            0,
            "auto",
          ]
        }
        height={0}
        hide={false}
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Tooltip
        active={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        contentStyle={Object {}}
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        isAnimationActive={true}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        separator=" : "
        trigger="hover"
        useTranslate3d={false}
        viewBox={
          Object {
            "x1": 0,
            "x2": 0,
            "y1": 0,
            "y2": 0,
          }
        }
        wrapperStyle={Object {}}
      />
      <Legend
        align="center"
        iconSize={14}
        layout="horizontal"
        verticalAlign="bottom"
      />
      <Line
        activeDot={true}
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={false}
        dataKey="worst"
        dot={true}
        fill="#fff"
        hide={false}
        isAnimationActive={true}
        legendType="line"
        points={Array []}
        stroke="#FF7A00"
        strokeWidth={1}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Line
        activeDot={true}
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={false}
        dataKey="average"
        dot={true}
        fill="#fff"
        hide={false}
        isAnimationActive={true}
        legendType="line"
        points={Array []}
        stroke="#6F52ED"
        strokeWidth={1}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
      <Line
        activeDot={true}
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={false}
        dataKey="best"
        dot={true}
        fill="#fff"
        hide={false}
        isAnimationActive={true}
        legendType="line"
        points={Array []}
        stroke="#28E451"
        strokeWidth={1}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
    </LineChart>
  </ResponsiveContainer>
</div>
`;
